app/renderer/util.js

```diff
@@ -114,69 +120,124 @@ function isXpathUnique(xpath, doc, domNode) {
   return [true];
 }
 
+/**
+ * Given an xml doc and a current dom node, try to find a robust xpath selector qualified by
+ * key attributes, which is unique in the document (or unique plus index).
+ *
+ * @param {string} xpath
+ * @param {DOMDocument} doc
+ * @param {Array<string>|Array<[string, string]>} attrs - a list of attributes to consider, or
+ * a list of pairs of attributes to consider in conjunction
+ * @param {boolean} isPairs - whether we are looking for single attributes or a conjunction of
```

Comment:
`[isPairs=false]` is the proper format

Reply:
also, we probably don't need this arg as we could still dynamically detect the format of each item